praw/reddit.py

```diff
@@ -183,25 +183,47 @@ def request(self, *args, **kwargs):
         self._unique_counter = 0
         self._validate_on_submit = False
 
+        if config is None:
+            config = get_praw_config()
+        section_name = site_name or config.default_section
         try:
-            config_section = site_name or os.getenv("praw_site") or "DEFAULT"
-            self.config = Config(
-                config_section, config_interpolation, **config_settings
-            )
-        except configparser.NoSectionError as exc:
-            help_message = (
-                "You provided the name of a praw.ini "
-                "configuration which does not exist.\n\nFor help "
-                "with creating a Reddit instance, visit\n"
-                "https://praw.readthedocs.io/en/latest/code_overvi"
-                "ew/reddit_instance.html\n\n"
-                "For help on configuring PRAW, visit\n"
-                "https://praw.readthedocs.io/en/latest/getting_sta"
-                "rted/configuration.html"
-            )
-            if site_name is not None:
-                exc.message += "\n" + help_message
-            raise
+            section = config[section_name]
+        except KeyError:
+            msg = (f"No section {section_name!r}"
+                '''
+
+You provided the name of a praw.ini configuration which does not exist.
+
+For help with creating a Reddit instance, visit
+https://praw.readthedocs.io/en/latest/code_overview/reddit_instance.html
+
+For help on configuring PRAW, visit
+https://praw.readthedocs.io/en/latest/getting_started/configuration.html
+''')
+            class StrReprStr(str):
+                def __repr__(self):
+                    return str(self)
+            raise KeyError(StrReprStr(msg)) from None
+
+        settings = {}
+        settings.update(get_praw_environment_settings())
+        settings = {
```

Comment:
```suggestion
        config_settings = {
```

Reply:
That name is taken.

Reply:
Wait, then why do you bother making settings as an empty dict, then updating it, then overwriting it?

Reply:
Oops, I goofed up.